File: sortowanieBabelkowe.py



def sortowanieBabelkowe(lista):
    for i in range(len(lista)):
        for j in range(len(lista) - 1 - i):
            if lista[j] > lista[j + 1]:
                lista = swap(lista,j,j+1)
            printLista(lista)

def swap(lista, i, j):
    lista[i], lista[j] = lista[j], lista[i]
    return lista

def printLista(lista):
    for x in lista:
        print(x, end=', ')
    print(' ')



lista = [6, 1, 8, 9, 12, 3, -1, 0, -69]
sortowanieBabelkowe(lista)

File: sortowanie babelkowe.js
lista = [1, 6, 8, 9, 12, 3, -1, 0, -69]

for i in range(len(lista)):
    for j in range(len(lista) - 1):
        if lista[j] > lista[j+1]:
            temp = lista[j]
            lista[j] = lista[j+1]
            lista[j + 1] = temp
     for x in lista:
        print(x, end=', ')
     print('')
